[IF] http -> https 리다이렉트

kioku714 opened the pull request:
#12  http -> https 리다이렉트
~~#13 도메인 주소 동적 변경 - api.javacafe.co.kr 추가~~

----
2019-05-09 수정작업

- README.md: docker 실행 명령어에 볼륨으로 pem 키 경로 추가.
- AdminWebServer/default.conf: 애플리케이션 분기 처리 nginx를 호스트에 설치하는 것으로 결정되어 관련 설정 정보 삭제.
- WebServer/Dockerfile: nginx 설정파일 복사 코드 추가.
- WebServer/Dockerfile.dev: nginx 설정파일 복사 코드 추가.
- WebServer/default.conf: nginx 설정파일 추가.

kioku714 committed:
#12 http -> https 리다이렉트

kioku714 committed:
#13 [IF] 도메인 주소 동적 변경
- api.javacafe.co.kr 추가

rygh4775 commented:
@kioku714 
Nginx가 없는 API 서버 같은 경우 https 인증서 적용을 어떻게 하실 생각이세요?
(ELB 쪽에 https에 인증서 적용을 하는 것도 한 방법이 될것 같습니다.)

kioku714 commented:
@rygh4775 
ELB는 유료 서비스([Elastic Load Balancing 요금](https://aws.amazon.com/ko/elasticloadbalancing/pricing/))라서 다른 방법을 찾아봤는데 node.js 쪽에서 설정하면 해결될 것 같습니다. 이 PR은 클로즈 하고, 이슈 새로 등록해서 작업 진행하면 될 것 같습니다.

rygh4775 commented:
Admin webserver 쪽에 nginx 분기를 하고 있는데요...
이슈가 있을 것 같습니다.

### 저희 구조를 보면...
api.javacafe.co.kr:80 -> EC2:80 -> api-Docker:3000
admin.javacafe.co.kr:80 -> EC2:80 -> admin-Docker:8080
javacafe.co.kr:80 -> EC2:80 -> landing-Docker:8080
group1.javacafe.co.kr:80 -> EC2:80 -> frontend-Docker:8080
group2.javacafe.co.kr:80 -> EC2:80 -> frontend-Docker:8080 (위와 같은 도커 컨테이너)

### 위 구조가 불가능한 이유
EC2:80을 여러 Docker 포트에 맵핑 할수 없기 때문에 80포트의 Nginx는 1대만 떠야하는 상황

현재 PR을 보면 Host:80포트를 컨트롤하는 Nginx는 AdminWebser에 있고,
애플리케이션 분기 처리를 AdminWebserver에 있는 Nginx에서 전부 다 해야되는 상황 같습니다.

@kioku714 
제가 이해하고 있는게 맞나요?

kioku714 commented:
음.. 생각해보니 애플리케이션 분기 처리용 nginx  docker container를 추가하는 편이 나을 것 같네요.
@rygh4775 어떻게 생각하시나요?

rygh4775 commented:
@kioku714 
네 말씀하신 방법도 한가지 방법이 될 수 있을 것 같습니다.
다만, Docker로 진행하는게 운영 상에 더 불편함을 줄수 있을 것 같다는 생각이드는데요..

Nginx와 관련된 운영성 작업은 크게 두가지 일 것 같습니다.
1. Nginx 설치
> 일회성 작업 (시스템 확장시 일회성 작업을 반복)
2. Nignx 설정 파일 수정 및 재시작
> 다소 여러 번 수행 필요 (Nginx에 설정에 Elasticsearch-plugin 설정도 추가되어야함)

2번의 경우 Nginx conf파일은 Volume 마운트를 설정하고 Host에서 수정을 하면 되지만,
재시작 할 때 불편함이 있습니다.
- container를 재시작 (다운타임 발생)
- container에 접속하여 Nginx를 재시작 (다운타임 0에 가까움) 

위 두가지 방법 중 한가지로 재시작을 할 수 있을 것 같습니다.

애플리케이션 분기와 인증서 설정을 담당하는 Nginx 같은 경우에는 Dockerizing 하지 않고 Host에 설치하는게 관리 측면에서 더 좋지 않을까 싶습니다.

cc. @clghks

kioku714 committed:
#12 http -> https 리다이렉트
- README.md: 볼륨으로 pem 키 경로 추가
- AdminWebServer/default.conf: 애플리케이션 분기 처리 nginx를 호스트에 설치하는 것으로 결정되어 관련 설정 정보 삭제.
- WebServer/Dockerfile: nginx 설정파일 복사 코드 추가
- WebServer/Dockerfile.dev: nginx 설정파일 복사 코드 추가
- WebServer/default.conf: nginx 설정파일 추가

rygh4775 reviewed (changes requested):
이제 Host에 설치된 nginx에서 https 관련 처리를 하기때문에
Docker 컨테이너에 설치되는 nginx에서 https 관련 처리는 필요 없어 보이네요.

kioku714 committed:
fix: https 설정 제거, docker 실행파일에서 ssl pem키 볼륨 제거
